contracts/ClaimsManagerWithKlerosArbitration.sol

```diff
@@ -148,6 +148,31 @@ contract ClaimsManagerWithKlerosArbitration is
                 ],
             "Invalid claim-dispute pair"
         );
+        // Ruling options
+        // 0: Kleros refused to arbitrate. We allow both parties to appeal to this.
+        // 1: Pay the claim. Only the mediator can appeal to this.
+        // 2: Pay the settlement. Only the claimant can appeal to this.
+        // We don't check the dispute status (if it's appealable), as the appeal() call
+        // below should revert in that case anyway.
+        if (msg.sender == claims[claimIndex].claimant) {
+            require(
+                arbitrator.currentRuling(disputeId) != 1,
+                "Ruling agrees with claimant"
+            );
+        } else if (
+            manager == msg.sender ||
+            IAccessControlRegistry(accessControlRegistry).hasRole(
+                mediatorRole,
+                msg.sender
+            )
+        ) {
+            require(
+                arbitrator.currentRuling(disputeId) != 2,
+                "Ruling agrees with mediator"
+            );
+        } else {
```

Comment:
This is just a my personal preference but I find this easier to read:

```suggestion
        if (arbitrator.currentRuling(disputeId) == 1) {
            require(manager == msg.sender ||
            IAccessControlRegistry(accessControlRegistry).hasRole(
                mediatorRole,
                msg.sender
            ), "Only mediator can appeal pay claim" );
        } else if (arbitrator.currentRuling(disputeId) == 2) {
            require(msg.sender == claims[claimIndex].claimant,
                "Only claimant can appeal pay settlement"
            );
        } else {
```

Reply:
Kinda difficult to implement the else though

Reply:
You are right. Didn't think about the `else` block 🤦🏻  but maybe we can get rid of it and just create a modifier `onlyAppealer` where we check that `msg.sender` is either claimant or manager/mediator. 

Again, I believe this suggestion makes the code easier to read but maybe I'm wrong. It's your call

Reply:
I'll keep it as is because it's more gas efficient this way